src/main/java/org/sagebionetworks/bridge/notification/worker/BridgeNotificationWorkerProcessor.java

```diff
@@ -410,36 +410,39 @@ private void notifyUser(String studyId, StudyParticipant participant, Notificati
         WorkerConfig workerConfig = dynamoHelper.getNotificationConfigForStudy(studyId);
 
         // Get notification messages for type.
-        Map<String, String> messagesByDataGroup;
+        List<String> messageList = null;
         switch (notificationType) {
             case CUMULATIVE:
-                messagesByDataGroup = workerConfig.getMissedCumulativeActivitiesMessagesByDataGroup();
+                messageList = workerConfig.getMissedCumulativeActivitiesMessagesList();
                 break;
             case EARLY:
-                messagesByDataGroup = workerConfig.getMissedEarlyActivitiesMessagesByDataGroup();
+                messageList = workerConfig.getMissedEarlyActivitiesMessagesList();
                 break;
             case LATE:
-                messagesByDataGroup = workerConfig.getMissedLaterActivitiesMessagesByDataGroup();
+                messageList = workerConfig.getMissedLaterActivitiesMessagesList();
                 break;
             case PRE_BURST:
-                messagesByDataGroup = workerConfig.getPreburstMessagesByDataGroup();
+                // Narrow down notification messages for data group.
+                Map<String, List<String>> messagesByDataGroup = workerConfig.getPreburstMessagesByDataGroup();
+                for (String oneDataGroup : participant.getDataGroups()) {
+                    if (messagesByDataGroup.containsKey(oneDataGroup)) {
+                        messageList = messagesByDataGroup.get(oneDataGroup);
+                        break;
+                    }
+                }
                 break;
             default:
                 throw new IllegalStateException("Unexpected type " + notificationType);
         }
 
-        // Narrow down notification messages for data group.
-        String message = null;
-        for (String oneDataGroup : participant.getDataGroups()) {
-            if (messagesByDataGroup.containsKey(oneDataGroup)) {
-                message = messagesByDataGroup.get(oneDataGroup);
-                break;
-            }
-        }
-        if (message == null) {
+        if (messageList == null) {
             throw new IllegalStateException("No messages found for type " + notificationType + " for user " + userId);
         }
 
+        // Pick message at random.
+        int randomIndex = rng.nextInt(messageList.size());
+        String message = messageList.get(randomIndex);
```

Comment:
As someone who makes a lot of generators, pure randomness sometimes produces very annoying results, like the same message 12 times. Rotating through the messages per user might be better for the user, but requires having some state so probably isn't worth it here.